server/lib/availabilitySync.ts

```diff
@@ -727,74 +744,77 @@ class AvailabilitySync {
+    season: Season,
+    showExists: boolean,
+    showExists4k: boolean
+  ): Promise<void> {
     const ratingKey = media.ratingKey;
     const ratingKey4k = media.ratingKey4k;
 
-    let seasonExistsInPlex = false;
-    let seasonExistsInPlex4k = false;
+    let seasonExistsInPlex = true;
+    let seasonExistsInPlex4k = true;
+    let plexSeason: PlexMetadata | undefined;
+    let plexSeason4k: PlexMetadata | undefined;
 
-    try {
-      if (ratingKey) {
+    // Check each plex instance to see if the season exists
+    if (ratingKey && showExists) {
+      try {
         const children =
           this.plexSeasonsCache[ratingKey] ??
           (await this.plexClient?.getChildrenMetadata(ratingKey)) ??
           [];
         this.plexSeasonsCache[ratingKey] = children;
-        const seasonMeta = children?.find(
+        plexSeason = children?.find(
           (child) => child.index === season.seasonNumber
         );
-
-        if (seasonMeta) {
-          seasonExistsInPlex = true;
-        }
+      } catch (ex) {
+        logger.debug(`Failed to retrieve Plex's non-4k season metadata.`, {
+          errorMessage: ex.message,
+          label: 'AvailabilitySync',
+        });
       }
-      if (ratingKey4k) {
+    }
+    if (ratingKey4k && showExists4k) {
+      try {
         const children4k =
           this.plexSeasonsCache[ratingKey4k] ??
           (await this.plexClient?.getChildrenMetadata(ratingKey4k)) ??
           [];
         this.plexSeasonsCache[ratingKey4k] = children4k;
-        const seasonMeta4k = children4k?.find(
+        plexSeason4k = children4k?.find(
           (child) => child.index === season.seasonNumber
         );
-
-        if (seasonMeta4k) {
-          seasonExistsInPlex4k = true;
-        }
-      }
-    } catch (ex) {
-      if (!ex.message.includes('response code: 404')) {
-        logger.debug(`Failed to retrieve plex's children metadata`, {
+      } catch (ex) {
+        logger.debug(`Failed to retrieve Plex's 4k season metadata.`, {
           errorMessage: ex.message,
           label: 'AvailabilitySync',
         });
       }
     }
+
+    if (!plexSeason) {
+      seasonExistsInPlex = false;
+    }
+
+    if (!plexSeason4k) {
+      seasonExistsInPlex4k = false;
+    }
+
     // Base case if both season versions exist in plex
     if (seasonExistsInPlex && seasonExistsInPlex4k) {
-      return true;
+      return;
     }
 
-    const existsInSonarr = await this.seasonExistsInSonarr(
+    await this.seasonExistsInSonarr(
       media,
       season,
       seasonExistsInPlex,
-      seasonExistsInPlex4k
+      seasonExistsInPlex4k,
+      showExists,
+      showExists4k
     );
-
-    if (existsInSonarr) {
-      logger.warn(
-        `Season ${season.seasonNumber}, media ID ${media.id} exists in at least one Sonarr or Plex instance. Media will be updated if set to available.`,
-        {
-          label: 'AvailabilitySync',
-        }
-      );
-
-      return true;
-    }
-
-    return false;
```

Comment:
We do not have to return a boolean here anymore as we have now nested the season checker within the sonarr checker.